content/authors/client-side-validation-with-react-alert/index.md

```diff
@@ -0,0 +1,366 @@
+
+The structure above shows that `checks.js` is inside the `utils` folder. `checks.js` is where we’d have all the client-side validation logic.
+
+Since we’d be interacting/writing this logic for web forms... we need to look for a way to target the input elements that are receiving data from the user at the client side. Luckily for
+us, the browser has provided some DOM APIs that we can use to achieve that. Take a look at an example below:
+
+```html
+<input type="”text”" name="”fullname”" id="”fullname”" className="”fullname”" />
+```
+
+```js
+const fullnameInputField = document.querySelector(“#fullname”)
+
+console.log(fullnameInputField)
+```
+
+The snippet above logs the HTML code to the console. You can check the result by open the developer tools of your browser, on Google chrome, this shortcut brings the console up. `ctrl + shift + i`.
+
+If we were to target the input element via its `className`, the snippet above would now be:
+
+```js
+const fullnameInputField = document.querySelector(“.fullname”)
+```
+
+Note that the pound sign `(#)` has changed to the period sign `(.)` because we’re targeting that DOM node via its className property.
+
+Now let’s have a look at the validation snippet below:
+
+```js
+let errMsg;
+
+const validateSignUp = (email, password, confirmPassword, alert) => {
+  // targeting all form fields
+  const emailInput = document.querySelector("#email");
+  const passwordInput = document.querySelector("#pwd");
+  const confirmPasswordInput = document.querySelector("#pwd_conf");
+
+  if (!email) {
+    alert.error("Please input your email address");
+    emailInput.focus();
+    errMsg = false;
+  } else if (password === "") {
+    alert.error("Please provide your password!");
+    passwordInput.focus();
+    errMsg = false;
+  } else if (password.value <= 7) {
+    alert.error("Your password must have 8 characters or greater");
+    passwordInput.focus();
+    errMsg = false;
+  } else if (
+    typeof password !== "undefined" &&
+    typeof confirmPassword !== "undefined"
+  ) {
+    if (password != confirmPassword) {
+      alert.error("Passwords don't match");
+      confirmPasswordInput.focus();
+      errMsg = false;
+    }
+  } else {
+    alert.success("You've signed up successfully. Proceed to login");
+    errMsg = true;
+  }
+  return errMsg;
+};
+
+export default signUpCheck;
+```
+
+Wow! I know that might be your reaction right now, but do not fret, We’d have a breakdown of what each snippet does as we move on.
+
+- **Passing form values as props**
+  The snippet above is a helper function that performs validation on the type of data that gets into the form field before it is sent to the backend server. The function accepts the `email`, `password`, `confirmPassword`, and `alert` as arguments.
+
```

Comment:
>  The snippet above is a helper function that performs validation on the input form data before the data gets sent to the backend server. The function accepts the `email`, `password`, `confirmPassword`, and `alert` as arguments.

I feel this should be...

 The snippet above is a helper function that performs validation on the input form data before **it** gets sent to the backend server. The function accepts the `email`, `password`, `confirmPassword`, and `alert` as arguments.
 
 To avoid repitions

Reply:
I've fixed it at my end